rust/xaynet-core/src/mask/masking.rs

```diff
@@ -122,12 +145,22 @@ impl Aggregation {
             return Err(UnmaskingError::NoModel);
         }
 
-        if self.nb_models > self.object.config.model_type.max_nb_models() {
+        if self.nb_models > self.object.vector.config.model_type.max_nb_models() {
             return Err(UnmaskingError::TooManyModels);
         }
 
-        if self.object.config != mask.config || self.object_size != mask.data.len() {
-            return Err(UnmaskingError::MaskMismatch);
+        if self.nb_models > self.object.scalar.config.model_type.max_nb_models() {
+            return Err(UnmaskingError::TooManyScalars);
```

Comment:
Can `self.object.vector.config.model_type.max_nb_models()` and `self.object.scalar.config.model_type.max_nb_models()` actually differ? Seems like the max is actually capped to the min of these values. More generally, can the scalar masking config and model masking config have incompatibilities?

Reply:
I agree this could probably done in one check. more generally, since the scalar could be masked with a different config to weights, we should do the same check to make sure it admits all of them for aggregating. will address separately.